source/solvers/heat_transfer_assemblers.cc

```diff
@@ -786,24 +788,142 @@ HeatTransferAssemblerLaser<dim>::assemble_rhs(
               local_rhs(i) += laser_heat_source * phi_T_i * JxW;
             }
+        } // end loop on quadrature points
+    }
+}
+
+template class HeatTransferAssemblerLaserExponentialDecay<2>;
+template class HeatTransferAssemblerLaserExponentialDecay<3>;
+
+template <int dim>
+void
+HeatTransferAssemblerLaserMaterialInterfaceVOF<dim>::assemble_matrix(
+  HeatTransferScratchData<dim> & /*scratch_data*/,
+  StabilizedMethodsCopyData & /*copy_data*/)
+{}
 
+template <int dim>
+void
+HeatTransferAssemblerLaserMaterialInterfaceVOF<dim>::assemble_rhs(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  // Laser parameters
+  const double concentration_factor = laser_parameters->concentration_factor;
+  const double laser_power          = laser_parameters->laser_power;
+  const double absorptivity         = laser_parameters->laser_absorptivity;
+  const double laser_start_time     = laser_parameters->start_time;
+  const double laser_end_time       = laser_parameters->end_time;
+  const double beam_radius          = laser_parameters->beam_radius;
+  const double current_time = this->simulation_control->get_current_time();
+
+  if (current_time >= laser_start_time && current_time <= laser_end_time)
+    {
+      // Get laser path
+      Function<dim> &laser_scan_path = *(laser_parameters->laser_scan_path);
+      laser_scan_path.set_time(current_time);
+
+      // For the laser heat source calculations, we need the radial distance, r,
+      // (in a dim-1 dimensional plane perpendicular to the laser beam
+      // direction) between the laser focal point and the quadrature points.
+      // Hence, we get the laser location (laser_location) as a Point<dim>, in
+      // which the first and second components show the position of the laser
+      // focal point in a plane perpendicular to the emission direction, and the
+      // (dim-1)th component denotes the position of the laser focal point in
+      // the direction of emission. Then we use dim-1 auxiliary variables
+      // (Point<dim-1> laser_location_on_surface) to store the position of the
+      // laser focal point in the perpendicular plane to the emission direction.
+
+      // Get laser location
+      Point<dim> laser_location;
+      laser_location[0] = laser_scan_path.value(
+        laser_location, laser_parameters->perpendicular_plane_coordinate_one);
+      if constexpr (dim == 3)
+        {
+          laser_location[1] = laser_scan_path.value(
+            laser_location,
+            laser_parameters->perpendicular_plane_coordinate_two);
+        }
+      laser_location[dim - 1] =
+        laser_scan_path.value(laser_location,
+                              laser_parameters->beam_orientation_coordinate);
+
+      // Get laser location on the operation surface
+      Point<dim - 1> laser_location_on_surface;
+      for (unsigned int d = 0; d < dim - 1; d++)
+        laser_location_on_surface[d] = laser_location[d];
```

Comment:
So in this model you still specify the laser focal point? @hepap do you think that'S the right thing? I wold have thought we would specify something like just the laser vector and a point on that vector and assume automatically that the focal point was the intersection between the surface and that laser vector. AKA no focal point. Otherwise it's tricky no?

Reply:
I would assume that for the surface flux, we only need the distance  between the "laser vector" and the current quadrature point in think, while in the exponential decay the depth is required.

I think it is the same as saying that the 2D gaussian distribution is the same no matter the "depth" of the interface.

I will check that in more details tomorrow, but I think the implementation bellow is still good.